### PR TITLE
[BugFix]  When replica enter decommission,  transaction will nerver complete.

### LINKED ISSUE #52769: When the server machine is not recovered,  transaction will nerver complete
### Steps to reproduce the behavior (Required)
```
// In order for the transaction to complete in time for this scenario: the server machine is not recovered.
        // 1. Transaction TA writes to a two-replicas tablet and enters the committed state.
        //    The tablet's repliace are replicaA, replicaB.
        // 2. replicaA, replicaB generate tasks: PublishVersionTaskA, PublishVersionTaskB.
        //    PublishVersionTaskA/PublishVersionTaskB successfully submitted to the beA/beB via RPC.
        // 3. The machine where beB is located hangs and is not recoverable.
        //   Therefore PublishVersionTaskA is finished,PublishVersionTaskB is unfinished.
        // 4. FE clone replicaC from replicaA, BE report replicaC info.
        //   But transactions must rely on replicaA and replicaB, so TA will block on committed.
```
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by zhangheihei**
## Why I'm doing:
 1. Transaction TA writes to a two-replicas tablet and enters the committed state. The tablet's replica are replicaA, replicaB.
 2. replicaA, replicaB generate tasks: PublishVersionTaskA, PublishVersionTaskB. PublishVersionTaskA/PublishVersionTaskB successfully submitted to the beA/beB via RPC.
 3. The machine where beB is located hangs and is not recoverable. Therefore PublishVersionTaskA is finished,PublishVersionTaskB is unfinished.
 4. FE clone replicaC from replicaA, BE report replicaC info.   But transactions must rely on replicaA and replicaB, so TA will block on committed.
## What I'm doing:

Fixes #52769

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java [hunk 1]
```diff
@@ -123,8 +126,9 @@ public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInf
                         long lastFailedVersion = replica.getLastFailedVersion();
                         long newVersion = version;
                         long lastSucessVersion = replica.getLastSuccessVersion();
-                        if (!txnState.tabletCommitInfosContainsReplica(tablet.getId(), replica.getBackendId(),
-                                replica.getState())
+                        if ((!txnState.tabletCommitInfosContainsReplica(tablet.getId(), replica.getBackendId(),
+                                replica.getState()) && !txnState.
+                                checkTransactionDependencyReplicasNotCommited((LocalTablet) tablet, quorumReplicaNum))
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java:131]: in this case, replicaC does not have the new version rowset but the version is updated to newVersion, this is not right?

**Review by luohaha — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java [hunk 2]
```diff
@@ -127,9 +127,9 @@ public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInf
                         if (!txnState.tabletCommitInfosContainsReplica(tablet.getId(), replica.getBackendId(),
                                 replica.getState())
                                 || errorReplicaIds.contains(replica.getId())) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java:129]:
```suggestion
                        if (transactionState.checkReplicaNeedSkip(tablet, replica, partitionCommitInfo)
                                || errorReplicaIds.contains(replica.getId())) {
```

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java:129]: better to not set `lastFailedVersion` on the cloned healthy replica

Comment (zhangheihei) [reply fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java]: tks review

**Review by luohaha — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.4

**Pull request merged**